doc/design/state-machine.md

```diff
@@ -0,0 +1,185 @@
+# HiveD State Machines
```

Comment:
Link doc/design to the first page readme, here:
## Doc
1. [User Manual](doc/user-manual.md)
2. [Feature Demo](example/feature/README.md)